libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -407,3 +410,35 @@ void ConstantEvaluator::endVisit(TupleExpression const& _tuple)
 	if (!_tuple.isInlineArray() && _tuple.components().size() == 1)
 		m_values[&_tuple] = evaluate(*_tuple.components().front());
 }
+
+void ConstantEvaluator::endVisit(FunctionCall const& _functionCall)
+{
+	using util::keccak256;
+	using util::h256;
+
+	auto const* builtinFunction = dynamic_cast<MagicVariableDeclaration const*>(ASTNode::referencedDeclaration(_functionCall.expression()));
+	if (!builtinFunction)
+		return;
+
+	auto const* functionType = builtinFunction->functionType(true);
+	solAssert(functionType);
+	switch (functionType->kind())
+	{
+		case FunctionType::Kind::ERC7201:
+		{
+			solAssert(_functionCall.arguments().size() == 1);
```

Comment:
Shouldn't we report an error instead?

Reply:
That is already checked in `TypeChecker::typeCheckFunctionGeneralChecks`.
I think also it doesn't make sense to raise such error here.
But your question touches on something I am uncomfortable still in extending the `ConstantEvaluator`, which is depending/assuming that analysis step already passed.